libraries/CurieAudio/src/I2SController.cpp

```diff
@@ -0,0 +1,76 @@
+//***************************************************************
+//
+// Copyright (c) 2016 Intel Corporation.  All rights reserved.
+//
+// This library is free software; you can redistribute it and/or
+// modify it under the terms of the GNU Lesser General Public
+// License as published by the Free Software Foundation; either
+// version 2.1 of the License, or (at your option) any later version.
+//
+// This library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+// Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public
+// License along with this library; if not, write to the Free Software
+// Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
+//
+//***************************************************************
+
+#include "I2SController.h"
+#include "variant.h"
+
+uint16_t sampleRateLookupTable[MAX_I2S_SIZE * MAX_I2S_RATE] = {
+  //  8K  12k  22k  24k  44k  48k
+    125,  83,  45,  42,  23,  21,    // I2S_16bit
+     83,  56,  30,  28,  15,  14,    // I2S_24bit
+     63,  42,  23,  21,  11,  10     // I2S_32bit
+};
+
+uint8_t sampleSizeLookupTable[MAX_I2S_SIZE] = {16, 24, 32};
+
+
+I2SController::I2SController()
+{
+  sampleRateMap = sampleRateLookupTable;
+  sampleSizeMap = sampleSizeLookupTable;
+}
+
+
+void I2SController::muxTX(bool enable)
+{
+    int mux_mode = GPIO_MUX_MODE;
+    if(enable)
+    {
+        mux_mode = I2S_MUX_MODE;
+    }
+    
+    /* Set SoC pin mux configuration */
+    SET_PIN_MODE(g_APinDescription[I2S_TXD].ulSocPin, mux_mode);
+    SET_PIN_MODE(g_APinDescription[I2S_TWS].ulSocPin, mux_mode);
+    SET_PIN_MODE(g_APinDescription[I2S_TSCK].ulSocPin, mux_mode);
+    g_APinDescription[I2S_TXD].ulPinMode = mux_mode;
```

Comment:
These lines (53-55) will throw a compile error after https://github.com/01org/corelibs-arduino101/pull/308 is merged
(same goes for any further modifications to g_APinDescription).

The above pull request has already undergone review & testing-- I suggest you review it yourself, and rebase once it is merged (which will probably be today).